### PR TITLE
Add documented DerivativeMultiPhaseMaterial example 

### LINKED ISSUE #4800: DerivativeMultiPhaseMaterial needs an example
actually @tonkmr needs it, but it will illustrate the ```DerivativeMultiPhaseMaterial``` class

### PR DESCRIPTION
**Pull request by dschwen**
There you go @jwpeterson , another one-commit PR!

Closes #4800.

### CODE REVIEW

File: modules/phase_field/examples/multiphase/DerivativeMultiPhaseMaterial.i [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+[GlobalParams]
+  # let's output all material properties for demonstration purposes
+  outputs = exodus
+
+  # prefactor on the penalty function kernels. The higher this value is, the
+  # more rigorously the constraint is enforced
+  penalty = 1e3
+[]
+
+#
```
Comment (tonkmr) [inline modules/phase_field/examples/multiphase/DerivativeMultiPhaseMaterial.i]: As soon as a user opens this in peacock and then saves it, all of the comments outside of the blocks will be stuck at the top of the input file. You need to move the comments describing a block inside of the block itself for peacock to not reformat it.

File: modules/phase_field/examples/multiphase/DerivativeMultiPhaseMaterial.i [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+  # parameters evolved using the ACMultiInterface kernel
+  #
+  [./cross_terms]
+    type = CrossTermGradientFreeEnergy
+    variable = cross_energy
+    interfacial_vars = 'eta1 eta2 eta3'
+    #
+    # The interface coefficient matrix. This should be symmetrical!
+    #
+    kappa_names = 'kappa11 kappa12 kappa13
```
Comment (tonkmr) [inline modules/phase_field/examples/multiphase/DerivativeMultiPhaseMaterial.i]: The comments describing an input parameter need to be on the same line as the parameter for them to show up in the right block in peacock

**Comment by tonkmr** [comment]
I'm fine if we just decide that we aren't formatting for peacock.

**Comment by dschwen** [comment]
Yeah, I'm voting for keeping it as it is. That way we can keep it more readable and users can just read the entire file in a text editor (I suggest https://atom.io/packages/language-moose).

**Pull request merged**